great_expectations/data_context/types/base.py

```diff
@@ -925,6 +944,18 @@ class DatasourceConfigSchema(Schema):
     class Meta:
         unknown = INCLUDE
 
+    # Note: name is optional currently to avoid updating all documentation within
+    # the scope of this work.
+    name = fields.String(
+        required=False,
+        allow_none=True,
+    )
+    id = fields.String(
```

Comment:
```suggestion
    id_ = fields.String(
```
Not sure how the class level `id` will interact with the object instance reserved `id` attribute.
But I would think this is safer.

Reply:
If I change to `id_` the `id` field does not show up in a `datasourceConfigSchema.dump()` even when setting `data_key="id"`. Trying to figure out the best approach here but very open to ideas!

Reply:
That's weird. Although there are people in this SO saying they had problems too.

https://stackoverflow.com/questions/44434180/how-to-serialize-a-marshmallow-field-under-a-different-name

Maybe we are using an earlier version of MM.
In which case we need to use `attribute="id"`.

Reply:
I was wrong about `.id`.

https://stackoverflow.com/questions/16153501/python-what-is-the-object-method-of-built-in-id